_posts/2019-06-04-loading-large-image-data-with-dask-array.md

```diff
@@ -0,0 +1,150 @@
+draft: true
+tags: [python,scipy,scikit-image,dask-image]
+theme: twitter
+---
+{% include JB/setup %}
+
+Series Overview
+---------------
+
+A common case in fields that acquire large amounts of imaging data is to write
+out smaller acquisitions into many small files. These files can tile a larger
+space, subsample from a larger time period, and may contain multiple channels.
+The acquistion techniques themselves are often state of the art and constantly
+pushing the envelope in term of how large a field of view can be acquired, at
+what resolution, and what quality (also known as signal-to-noise ratio).
+
+Once acquired this data presents a number of challenges. Algorithms often
+designed and tested to work on very small pieces of this data need to be scaled
+up to work on the full dataset. It might not be clear at the outset what will
+actually work and so exploration still plays a very big part of the whole
+process.
+
+Historically this analytical process has involved a lot of custom code. Often
+the analytical process is stitched together by a series of scripts possibly in
+several different languages that write various intermediate results to disk.
+Thanks to advances in modern tooling these process can be significantly
+improved. In this series of blogposts, we will outline ways for image
+scientists to leverage different tools to move towards a high level, friendly,
+cohesive, interactive analytical pipeline.
+
+Executive Summary
+-----------------
+
+This post explores a few simple workflows that a user might leverage to load
+their image with Dask. At the end of this post users should be able to
+construct a Dask Array, which they can use to further explore and manipulate
+their image data.
+
+Motivation
+----------
+
+Loading large image data can be a complex and often unique problem. Different
+groups may choose to store this across many files on disk, a commodity or
+custom database solution, or they may opt to store it in the cloud. Not all
+datasets within the same group may be treated the same for a variety of
+reasons. In short, this means loading data is a hard and expensive problem.
+
+Despite data being stored in different ways, often group want to reapply the
+same analytical pipeline to these datasets. However if the data pipeline is
+tightly coupled to a particular way of loading the data for later analytical
+steps, it may be very difficult if not impossible to reuse an existing
+pipeline. In other words, there is friction between the loading and analysis
+steps, which frustrates efforts to make things reusable.
+
+Having a modular and general way to load data makes it easy to present data
+stored differently in a standard way. Further having a standard way to present
+data to analytical pipelines allows that part of the pipeline to focus on what
+it does best, analysis! In general, this should decouple these to components in
+a way that improves the experience of users involved in all parts of the
+pipeline.
```

Comment:
I think that we probably need to reduce the exposition leading this post.  There are seven paragraphs of solid text before we get to non-text details like code, a table of results, or an image.  My experience is that most readers have a shorter attention span than this.

Some ideas:

1.  Drop the Series Overview section for now.  We can add something later after we have a series.
2.  Cut verbosity

    > Before: This post explores a few simple workflows that a user might leverage to load
their image with Dask.

    > After: We load image data with Dask.  
3.  Include a teaser code sample pretty much immediately afterwards in a "this is where we're going" comment, just so readers have a reason to stick around.

Reply:
Thanks for the feedback here.

1. I'd like to keep the series overview at least in this post as it will be the first one.
2. It probably has a fair bit of room for cutting (I have a tendency to be a bit verbose).
3. Happy to break this up with more code, pictures, diagrams, etc.